aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -256,13 +250,22 @@ at::Tensor PackedLinearWeightsQnnp::apply_dynamic_impl(at::Tensor input) {
   if (!input_scale.has_value()) {
     // Get the original weight and adjust it to uint8 from int8
     auto weight_contig = orig_weight;
-    int8_t* w_data = (int8_t*)weight_contig.data_ptr<c10::qint8>();
+
+    uint8_t* weight_zp_data = (uint8_t*)w_zero_points.data_ptr<c10::quint8>();
+    float* weight_scales_data = w_scales.data_ptr<float>();
+    requantization_scale =
+        generate_requantization_scales(w_scales, q_params.scale, 1.f);
+
+    // TODO Kimish, we are allocating affine_quantized regardless of per channel or not.
+    // This allocation is actually used only for packing weight and thus will be freed.
+    // Still we should be consistent. Fix this.
```

Comment:
Are the scale and offset even used here?  Could this just be a normal CPU tensor of dtype uint8?

Reply:
Yes it can be. Let me do that after this stack lands.